fs2/shared/src/main/scala/org/atnos/eff/addon/fs2/TaskEffect.scala

```diff
@@ -0,0 +1,229 @@
+package org.atnos.eff.addon.fs2
+
+import cats._
+import cats.implicits._
+import fs2._
+import org.atnos.eff._
+import org.atnos.eff.syntax.all._
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.TimeoutException
+import scala.util._
+
+case class TimedTask[A](task: (Strategy, Scheduler) => Task[A], timeout: Option[FiniteDuration] = None) {
+  def runNow(implicit strategy: Strategy, scheduler: Scheduler): Task[A] = timeout.fold(task(strategy, scheduler)) { t =>
+    for {
+      ref <- Task.ref[A]
+      _ <- ref.set(
+        (Task.fail(new TimeoutException).schedule(t): Task[A])
+          .race(task(strategy, scheduler))
+          .map(_.merge)
+      )
+      result <- ref.get
+    } yield result
+  }
+}
+
+object TimedTask {
+  final def TimedTaskApplicative: Applicative[TimedTask] = new Applicative[TimedTask] {
+    override def pure[A](x: A) = TimedTask((_, _) => Task.now(x))
+    override def ap[A, B](ff: TimedTask[(A) => B])(fa: TimedTask[A]) =
+      TimedTask((strategy, scheduler) =>
+        for {
+          fStarted <- Task.start(ff.runNow(strategy, scheduler))(strategy)
+          aStarted <- Task.start(fa.runNow(strategy, scheduler))(strategy)
+          fDone <- fStarted
+          aDone <- aStarted
+        } yield fDone(aDone)
+      )
+  }
+
+  implicit final def TimedTaskMonad: Monad[TimedTask] = new Monad[TimedTask] {
+    override def pure[A](x: A) = TimedTask((_, _) => Task.now(x))
+    override def flatMap[A, B](fa: TimedTask[A])(f: (A) => TimedTask[B]) =
```

Comment:
For the record I generally favour inserting new lines between methods unless they are really short definitions with almost similar signature or implementation, to help visualize the difference faster.